Storage/SQL/SQLiteRemoteClientsAndTabs.swift

```diff
@@ -316,38 +249,35 @@ open class SQLiteRemoteClientsAndTabs: RemoteClientsAndTabs {
     }
 
     open func getClientsAndTabs() -> Deferred<Maybe<[ClientAndTabs]>> {
-        var err: NSError?
-
         // Now find the clients.
-        let clientCursor = db.withConnection(&err) { connection, _ in
-            return connection.executeQuery("SELECT * FROM \(TableClients) WHERE EXISTS (SELECT 1 FROM \(TableRemoteDevices) rd WHERE rd.guid = fxaDeviceId) ORDER BY modified DESC", factory: SQLiteRemoteClientsAndTabs.remoteClientFactory)
+        let deferredClients = db.withConnection { connection -> [RemoteClient] in
+            let cursor = connection.executeQuery("SELECT * FROM \(TableClients) WHERE EXISTS (SELECT 1 FROM \(TableRemoteDevices) rd WHERE rd.guid = fxaDeviceId) ORDER BY modified DESC", factory: SQLiteRemoteClientsAndTabs.remoteClientFactory)
+            let clients = cursor.asArray()
+            cursor.close()
+            
+            return clients
         }
-
-        if let err = err {
-            clientCursor.close()
-            return deferMaybe(DatabaseError(err: err))
+        
+        guard let clients = deferredClients.value.successValue else {
+            return deferMaybe(DatabaseError(err: deferredClients.value.failureValue as NSError?))
         }
 
-        let clients = clientCursor.asArray()
-        clientCursor.close()
-
         log.debug("Found \(clients.count) clients in the DB.")
 
-        let tabCursor = db.withConnection(&err) { connection, _ in
-            return connection.executeQuery("SELECT * FROM \(TableTabs) WHERE client_guid IS NOT NULL ORDER BY client_guid DESC, last_used DESC", factory: SQLiteRemoteClientsAndTabs.remoteTabFactory)
+        var acc = [String: [RemoteTab]]()
+        let deferredTabsCursor = db.withConnection { connection -> Cursor<RemoteTab> in
+            connection.executeQuery("SELECT * FROM \(TableTabs) WHERE client_guid IS NOT NULL ORDER BY client_guid DESC, last_used DESC", factory: SQLiteRemoteClientsAndTabs.remoteTabFactory)
         }
-
-        log.debug("Found \(tabCursor.count) raw tabs in the DB.")
-
-        if let err = err {
+        
+        guard let tabCursor = deferredTabsCursor.value.successValue else {
```

Comment:
You're using `value` here because `withConnection` wraps in a `Deferred`. Just… put both queries inside the same `withConnection`, so we dump `TableTabs` and `TableClients` in one go, get a single `Deferred`, and then just call `.map` to turn the pair into `[ClientsAndTabs]`.